File: Dataset/Leetcode/valid/55/509.cpp
 class Solution {
public:
    bool XXX(vector<int>& nums) {
        int size = nums.size();
        int min = size - 1;
        for(int i=size - 1;i>=0;i--){
            if(i+nums[i]>=min) min = i;
        }
        if(!min) return true;
        return false;
    }
};

File: Dataset/Leetcode/valid/62/509.java
class Solution {
    public int XXX(int m, int n) {
int[][] a = new int[m][n];
        for (int i = 0; i < a.length; i++) {
            for (int j = 0; j < a[i].length; j++) {
                if(i==0||j==0){
                    a[i][j] = 1;
                }else {
                    a[i][j] = a[i-1][j]+a[i][j-1];
                }

            }
        }
        return a[m-1][n-1];
    }
}

